src/edu/harvard/lib/librarycloud/items/ItemDAO.java

```
@@ -0,0 +1,487 @@
+
+import javax.ws.rs.core.MultivaluedMap;
+import javax.xml.bind.JAXBElement;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import javax.xml.bind.Unmarshaller;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.stream.StreamResult;
+import javax.xml.transform.stream.StreamSource;
+
+import org.apache.log4j.Logger;
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.SolrQuery.ORDER;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.client.solrj.impl.HttpSolrServer;
+import org.apache.solr.client.solrj.impl.HttpSolrServer.RemoteSolrException;
+import org.apache.solr.client.solrj.response.FacetField;
+import org.apache.solr.client.solrj.response.QueryResponse;
+import org.apache.solr.common.SolrDocument;
+import org.apache.solr.common.SolrDocumentList;
+import org.dublincore.Metadata;
+
+/**
+ *
+ * ItemDAO makes queries to the solr index via solrj methods; item id info or
+ * query paramters are parsed and mapped into solr query syntax
+ *
+ * @author Michael Vandermillen
+ *
+ */
+public class ItemDAO {
+	Logger log = Logger.getLogger(ItemDAO.class);
+	private int limit = 10;
+	private Facet facet = null;
+
+	/**
+	 * Returns a MODS record for a given recordIdentifier.
+	 * 
+	 * @param id a recordIdentifier for a solr document
+	 * @return the ModsType for this recordidentifier
+	 * @see ModsType
+	 */
+	public ModsType getMods(String id) throws JAXBException {
+		SolrDocumentList docs;
+		SolrDocument doc;
+		ModsType modsType = new ModsType();
+		HttpSolrServer server = null;
+		try {
+			server = SolrServer.getSolrConnection();
+			SolrQuery query = new SolrQuery("recordIdentifier:" + id);
+			QueryResponse response = server.query(query);
+			docs = response.getResults();
+			if (docs.size() == 0)
+				throw new ResourceNotFoundException("Item " + id + " not found");
+			else {
+				doc = docs.get(0);

```

Should we also throw an error if ```docs.size() != 1``` ?